### PR TITLE
Migrate Clipboard to platform messages

### PR DESCRIPTION
**Pull request by abarth**
The platform plugin now supports clipboard operations.

### CODE REVIEW

**Comment by abarth** [comment]
@chinmaygarde @jason-simmons

**Review by chinmaygarde — APPROVED** [review]
lgtm

File: packages/flutter/lib/src/services/platform_messages.dart [hunk 1]
```diff
@@ -123,4 +129,35 @@ class PlatformMessages {
...
     });
   }
+
+  /// Sets a message handler that intercepts outgoing messages in binary form.
+  ///
+  /// The given callback will replace the currently registered callback (if any).
+  /// To remove the mock handler, pass `null` as the `handler` argument.
+  static void setMockBinaryMessageHandler(String name, Future<ByteData> handler(ByteData message)) {
+    if (handler == null)
+      _mockHandlers.remove(handler);
```
Comment (Hixie) [inline packages/flutter/lib/src/services/platform_messages.dart]: i think you meant to pass in `name` here

Comment (abarth) [inline packages/flutter/lib/src/services/platform_messages.dart]: Roger.  Folded into https://github.com/flutter/flutter/pull/6409

**Pull request merged**